instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/utils/idling/RouteAlternativesIdlingResource.kt

```diff
@@ -52,14 +58,24 @@ class RouteAlternativesIdlingResource(
         alternatives: List<DirectionsRoute>,
         routerOrigin: RouterOrigin
     ) {
+        if (!onRoutesAlternatives.invoke(routeProgress, alternatives, routerOrigin)) {
+            return
+        }
+
         // Verify this happens on the main thread.
-        if (Looper.myLooper() != Looper.getMainLooper()) {
-            calledOnMainThread = false
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            calledOnMainThread = true
         }
 
         mapboxNavigation.unregisterRouteAlternativesObserver(this)
         this.routeProgress = routeProgress
         this.alternatives = alternatives
+
         callback?.onTransitionToIdle()
     }
+
+    fun verifyOnRouteAlternativesPassed() {
```

Comment:
What does "passed" mean here? I'd think it's something like "the vehicle was following this route". But it seems like it's "alternative routes were received via callback". Can we change that to `verifyOnRouteAlternativesReceived` or something similar?